File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/UserResponse.java
package com.cinema.ticket_booking.data.model.response;

import com.google.gson.annotations.SerializedName;

public class UserResponse {
    @SerializedName("id")
    private String id;
    @SerializedName("email")
    private String email;
    @SerializedName("fullName")
    private String fullName;
    @SerializedName("phone")
    private String phone;
    @SerializedName("avatarUrl")
    private String avatarUrl;
    @SerializedName("role")
    private String role;
    @SerializedName("rewardPoints")
    private int cinePoints;
    @SerializedName("cinemaId")
    private String cinemaId;
    @SerializedName("cinemaName")
    private String cinemaName;

    @SerializedName("currentTierMinPoints")
    private Long currentTierMinPoints;
    @SerializedName("nextTierMinPoints")
    private Long nextTierMinPoints;
    @SerializedName("membershipTier")
    private String rank;

    @SerializedName("availableExp")
    private long availableExp;

    public long getAvailableExp() {
        return availableExp;
    }

    public String getRank() {
        return rank;
    }

    public String getId() {
        return id;
    }

    public String getEmail() {
        return email;
    }

    public String getFullName() {
        return fullName;
    }

    public String getPhone() {
        return phone;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public String getRole() {
        return role;
    }

    public int getCinePoints() {
        return cinePoints;
    }

    public String getCinemaId() {
        return cinemaId;
    }

    public String getCinemaName() {
        return cinemaName;
    }

    public Long getCurrentTierMinPoints() {
        return currentTierMinPoints;
    }

    public Long getNextTierMinPoints() {
        return nextTierMinPoints;
    }
}

File: App/app/src/main/java/com/cinema/ticket_booking/data/model/response/UserResponse.kt
package com.cinema.ticket_booking.data.model.response

import com.google.gson.annotations.SerializedName

data class UserResponse(
    @SerializedName("id")
    val id: String?,

    @SerializedName("email")
    val email: String?,

    @SerializedName("fullName")
    val fullName: String?,

    @SerializedName("phone")
    val phone: String?,

    @SerializedName("avatarUrl")
    val avatarUrl: String?,

    @SerializedName("role")
    val role: String?,

    @SerializedName("rewardPoints")
    val cinePoints: Int,

    @SerializedName("cinemaId")
    val cinemaId: String?,

    @SerializedName("cinemaName")
    val cinemaName: String?,

    @SerializedName("currentTierMinPoints")
    val currentTierMinPoints: Long?,

    @SerializedName("nextTierMinPoints")
    val nextTierMinPoints: Long?,

    @SerializedName("membershipTier")
    val rank: String?,

    @SerializedName("availableExp")
    val availableExp: Long,

    @SerializedName("allowMarketingNotification")
    private val allowMarketingNotificationField: Boolean?,

    @SerializedName("allowTransactionNotification")
    private val allowTransactionNotificationField: Boolean?
) {
    val allowMarketingNotification: Boolean
        @JvmName("getAllowMarketingNotification") get() = allowMarketingNotificationField ?: true

    val allowTransactionNotification: Boolean
        @JvmName("getAllowTransactionNotification") get() = allowTransactionNotificationField ?: true
}
